### PR TITLE
tests/subsys/settings/functional/file: fix nRF targets

### PR DESCRIPTION
**Pull request by nvlsianpu**
Pointed nRF flash driver for the NFFS.
Before improper flash driver name caused early test failure.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

### CODE REVIEW

**Review by rakons — APPROVED** [review]

File: tests/subsys/settings/functional/file/prj_nrf52840_pca10056.conf [hunk 1]
```diff
@@ -0,0 +1,26 @@
+CONFIG_ARM_MPU=n
```
Comment (ioannisg) [inline tests/subsys/settings/functional/file/prj_nrf52840_pca10056.conf]: Is this done for being able to write to flash?
If yes, have you considered the option that allows flash write?

Comment (carlescufi) [reply tests/subsys/settings/functional/file/prj_nrf52840_pca10056.conf]: @nvlsianpu can you reply?

Comment (nvlsianpu) [reply tests/subsys/settings/functional/file/prj_nrf52840_pca10056.conf]: @ioannisg Do you mean "MPU allow flash write option"?

Comment (nvlsianpu) [reply tests/subsys/settings/functional/file/prj_nrf52840_pca10056.conf]: I opened another PR for [select to allow flash write by MPU on nRF devices](https://github.com/zephyrproject-rtos/zephyr/pull/20314).

**Review by ioannisg — CHANGES_REQUESTED** [review]
@nvlsianpu are you going to update this PR? I would at least need the rationale behind disabling MPU.

**Comment by nvlsianpu** [comment]
@ioannisg Here you got:
This is functional test for the settings subsystem, no for MPU coexistence witch flash driver. I can keep MPU on and enable MPU_ALLOW_WRITE alike. What is your preference here?

**Comment by ioannisg** [comment]
> @ioannisg Here you got:
> This is functional test for the settings subsystem, no for MPU coexistence witch flash driver. I can keep MPU on and enable MPU_ALLOW_WRITE alike. What is your preference here?

Obviously, keep MPU on and enable MPU_ALLOW_FLASH_WRITE :)
And, please, do this for similar tests, if applicable.

**Comment by ioannisg** [comment]
With MPU on we are:
- able to run the test on user mode, if needed
- able to catch stack overflows, or any other memory violation issue

**Comment by nvlsianpu** [comment]
oki

**Comment by ioannisg** [comment]
@nvlsianpu any reason this does not execute on nrf9160?

**Comment by nvlsianpu** [comment]
Obviously this should work on nrf91 as well.

**Review by ioannisg — APPROVED** [review]

**Review by ioannisg — CHANGES_REQUESTED** [review]
sorry , changes not done yet ;)

**Comment by mbolivar** [comment]
@nvlsianpu do you plan on updating this? We are still carrying this patch downstream; it would be good to get this merged into mainline so we can have it for free when we merge next.

**Comment by nvlsianpu** [comment]
@ioannisg 

> @nvlsianpu any reason this does not execute on nrf9160?

Unfortunately I can't run this test on nrf9160 easily, moreover I think it doesn't make sens outside NCS multi-image infrastructure. Even in NCS storage usage on nrf9160 need some tweaks.

I would add nrf9160 here.

**Comment by nvlsianpu** [comment]
Wan't to keep PR as is.

**Comment by nvlsianpu** [comment]
Dosen't make sense as NFFS will be removed.
https://github.com/zephyrproject-rtos/zephyr/pull/21793

**Pull request closed without merging**